File: WorldFeed/Server/src/WorldFeed.Common/Public/Parameters/StreamsClient/CreateTweetStreamParameters.cs
namespace WorldFeed.Common.Public.Parameters.StreamsClient
{
    using WorldFeed.Common.Settings;

    public interface ICreateTweetStreamParameters : ICustomRequestParameters
    {
        /// <summary>
        /// Decide whether to use Extended or Compat mode
        /// </summary>
        TweetMode? TweetMode { get; set; }
    }

    public class CreateTweetStreamParameters : CustomRequestParameters, ICreateTweetStreamParameters
    {
        public TweetMode? TweetMode { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Public/Parameters/StreamsClient/CreateTweetStreamParameters.ts
import {CustomRequestParameters, ICustomRequestParameters} from "../CustomRequestParameters";
import {TweetMode} from '../../Settings/TweetinviSettings';

export interface ICreateTweetStreamParameters extends ICustomRequestParameters {
  // Decide whether to use Extended or Compat mode
  tweetMode?: TweetMode;
}

export class CreateTweetStreamParameters extends CustomRequestParameters implements ICreateTweetStreamParameters {
  public tweetMode?: TweetMode;
}
